crates/ruma-identifiers/src/server_name.rs

```diff
@@ -1,12 +1,32 @@
 //! Matrix-spec compliant server names.
 
+use std::net::Ipv4Addr;
+
 /// A Matrix-spec compliant server name.
 #[repr(transparent)]
 #[derive(PartialEq, Eq, PartialOrd, Ord, Hash)]
 pub struct ServerName(str);
 
 opaque_identifier_validated!(ServerName, ruma_identifiers_validation::server_name::validate);
 
+impl ServerName {
+    /// Returns the server name without the port.
+    pub fn without_port(&self) -> &str {
+        if let Some(end_of_ipv6) = self.0.find(']') {
```

Comment:
See my reviews in https://github.com/ruma/ruma/pull/811#discussion_r785306450, i think its best if you let the standard library parse these.